src/mpi/init/init_async.c

```diff
@@ -148,15 +149,38 @@ int MPIR_Init_async_thread(void)
     int mpi_errno = MPI_SUCCESS, thr_err;
     int global_rank, local_rank, local_size, threads_per_node;
     int *thread_affinity = NULL;
+    int num_cliques = 1, affinity_idx;
     MPIR_FUNC_TERSE_STATE_DECL(MPID_STATE_MPIR_INIT_ASYNC_THREAD);
 
     MPIR_FUNC_TERSE_ENTER(MPID_STATE_MPIR_INIT_ASYNC_THREAD);
 
+    /* Consider nodemap cliques when using debugging CVARs */
+    if (MPIR_CVAR_NUM_CLIQUES > 1) {
+        num_cliques = MPIR_CVAR_NUM_CLIQUES;
+    } else if (MPIR_CVAR_ODD_EVEN_CLIQUES) {
+        num_cliques = 2;
+    }
+
+    if (num_cliques > 1 && MPIR_CVAR_CH4_PROGRESS_THREAD_AFFINITY &&
+        strlen(MPIR_CVAR_CH4_PROGRESS_THREAD_AFFINITY) > 0) {
+        fprintf(stderr,
+                "Setting affinity for progress threads cannot work correctly with MPIR_CVAR_NUM_CLIQUES or MPIR_CVAR_ODD_EVEN_CLIQUES.\n");
```

Comment:
"cliques" is parsed and set in `src/util/mpir_pmi.c`. We should prevent having two separate places do the same logic -- single responsibility principle. Just let src/util/mpir_pmi.c expose a flag or better, expose a checking_routine, e.g. `MPIR_pmi_has_local_cliques()`.

Avoid complex logics, simply --
```
if (MPIR_pmi_has_local_cliques()) {
    [error exit]
}
```

Reply:
Done.